apps/emqx_mq/asn.1/ConsumerState.asn

```diff
@@ -0,0 +1,37 @@
+
+ConsumerState DEFINITIONS AUTOMATIC TAGS ::=
+BEGIN
+
+  StreamBufferProgress ::= SEQUENCE {
+    it OCTET STRING,
+    lastMessageId INTEGER OPTIONAL,
+    unacked SEQUENCE OF INTEGER
+  }
+
+  ShardProgress ::= SEQUENCE {
+    shard OCTET STRING,
+    progress CHOICE {
+      active ActiveShardProgress,
+      finished FinishedShardProgress
+    }
+  }
+
+  FinishedShardProgress ::= SEQUENCE {
+    generation INTEGER
+  }
+
+  ActiveShardProgress ::= SEQUENCE {
+    generation INTEGER,
+    stream OCTET STRING,
+    bufferProgress StreamBufferProgress
+  }
+
+  ConsumerStateV1 ::= SEQUENCE {
+    progress SEQUENCE OF ShardProgress
+  }
+
+  ConsumerState ::= CHOICE {
+    v1 ConsumerStateV1,
+    ...
```

Comment:
If you intend to make your schema extendable then it's almost always necessary to add tags manually.